.github/workflows/oas_pr.yml

```diff
@@ -0,0 +1,58 @@
+name: OAS PR build
+
+# This workflow is triggered on pull request events.
+# This runs tests and builds docs (but not deploy it).
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+
+    # OAS dependencies are specified in setup.py.
+    - name: Install OAS and its dependencies
+      run: |
+        python -m pip install --upgrade pip wheel
+        pip install .
+        pip install coverage coveralls testflo   # for tests and coveralls
+
+    # We need pySpline/pyGeo to run FFD tests.
+    - name: Install pySpline and pyGeo
+      run: |
+        sudo apt-get install openmpi-bin libopenmpi-dev
+        pip install mpi4py
+        git clone https://github.com/mdolab/pyspline.git
+        cd pyspline
+        cp config/defaults/config.LINUX_GFORTRAN.mk config/config.mk
+        make
+        pip install -e .
+        cd ..
+        git clone https://github.com/mdolab/pygeo.git
+        pip install -e ./pygeo
+
+    - name: Run tests
+      env:
+        OMPI_MCA_btl: ^openib   # prevent OpenMPI warning messages
+      run: |
+        cd openaerostruct
+        testflo -n 2 openaerostruct --coverage --coverpkg openaerostruct --cover-omit \*tests/\* --cover-omit \*docs/\*
+
+    # build docs (but not deployed)
+    - name: Build docs
+      run: |
+        cd openaerostruct/docs
+        make clean
+        make html
+
+    - name: Run coveralls
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Where is this stored? I don't see any secrets defined in the repo settings. I guess the token is needed to be able to comment on the PR after? Does it need any token to submit to Coveralls?

Reply:
I didn't understand very well, but `GITHUB_TOKEN` is a secret automatically created, so we don't need to define this by ourselves.
https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret

And we need this token to submit to coveralls.

Reply:
I see, that makes sense thanks.